storage/pruning/pruningStorer.go

```diff
@@ -264,6 +264,34 @@ func (ps *PruningStorer) Put(key, data []byte) error {
 	return ps.doPutInPersister(key, data, persisterToUse.getPersister())
 }
 
+// PutWithoutCache adds data to persistence medium and updates the bloom filter
+func (ps *PruningStorer) PutWithoutCache(key, data []byte) error {
+	ps.lock.RLock()
```

Comment:
Please split L269-L281 into a new function. You will remove some nested if branches

Reply:
Done.